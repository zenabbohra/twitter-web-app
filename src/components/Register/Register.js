import React, {Component} from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import styles from './Register.module.css';

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      registerName: '',
      registerEmail: '',
      registerPassword: ''
    }
  }


  onNameChange = (event) => {
    this.setState({registerName: event.target.value});
  };

  onEmailChange = (event) => {
    this.setState({registerEmail: event.target.value});
  };

  onPasswordChange = (event) => {
    this.setState({registerPassword: event.target.value});
  };

  onSignUpButtonClick = () => {
    const { registerName, registerEmail, registerPassword } = this.state;
    const { onRouteChange, loadUser } = this.props;
      fetch('http://circle.test/api/register', {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({
          name: registerName,
          email: registerEmail,
          password: registerPassword
        })
      })
        .then(response => response.json())
        .then(user => {
          if(user.name){
            onRouteChange('home');
            loadUser(user);
          }else {
            console.log(user);
          }
        })
        .catch(err => console.log(err))
  };

  render() {
    const { onNameChange, onEmailChange, onPasswordChange, onSignUpButtonClick } = this;
    return (
      <Container component="main" maxWidth="xs">
        <CssBaseline/>
        <div className={styles.paper}>
          <Avatar className={styles.avatar}>
            <LockOutlinedIcon/>
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign up
          </Typography>
            <Grid container spacing={2} >
              <Grid item xs={12}>
                <TextField
                  autoComplete="name"
                  name="name"
                  variant="outlined"
                  required
                  fullWidth
                  id="name"
                  label="Name"
                  autoFocus
                  onChange={onNameChange}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                  onChange={onEmailChange}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  variant="outlined"
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="current-password"
                  onChange={onPasswordChange}
                />
              </Grid>
            </Grid>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              className={styles.submit}
              onClick={onSignUpButtonClick}
            >
              Sign Up
            </Button>
            <Grid container className={styles.link}>
                <Link href="#" variant="body2" onClick={() => this.props.onRouteChange('sign in')}>
                  Already have an account? Sign in
                </Link>
            </Grid>
        </div>
      </Container>
    );
  }

}

export default Register;