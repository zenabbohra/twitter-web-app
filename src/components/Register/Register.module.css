.paper {
    margin-top: 64px;
    display: flex;
    flex-direction: column;
    align-items: center;
}
.avatar {
    margin: 8px;
    background-color: #3f51b5 !important;
}
.form {
    width: 100%;
    margin-top: 24px;
}
.submit {
    margin: 16px 0 16px !important;
}

.link {
    justify-content: center;
}