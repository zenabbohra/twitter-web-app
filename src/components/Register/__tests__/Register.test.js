import React from 'react';
import {shallow} from 'enzyme';
import Register from "../Register";

describe('<Register />', () => {
  describe('renders', () => {
    const subject = shallow(<Register/>);

    it('has two <TextField/>', () => {
      const textFields = subject.find('WithStyles(ForwardRef(TextField))');
      expect(textFields.length).toBe(3);
      expect(textFields).toMatchSnapshot();
    });

    it('has one <Button/>', () => {
      const button = subject.find('WithStyles(ForwardRef(Button))');
      expect(button.length).toBe(1);
      expect(button).toMatchSnapshot();
    });

    it('has one <Link/>', () => {
      const link = subject.find('WithStyles(ForwardRef(Link))');
      expect(link.length).toBe(1);
      expect(link).toMatchSnapshot();
    });
  });
});