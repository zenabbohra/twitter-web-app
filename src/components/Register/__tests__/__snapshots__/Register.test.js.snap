// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> renders has one <Button/> 1`] = `
<WithStyles(ForwardRef(Button))
  className="submit"
  color="primary"
  fullWidth={true}
  onClick={[Function]}
  type="submit"
  variant="contained"
>
  Sign Up
</WithStyles(ForwardRef(Button))>
`;

exports[`<Register /> renders has one <Link/> 1`] = `
<WithStyles(ForwardRef(Link))
  href="#"
  variant="body2"
>
  Already have an account? Sign in
</WithStyles(ForwardRef(Link))>
`;

exports[`<Register /> renders has two <TextField/> 1`] = `
Array [
  <WithStyles(ForwardRef(TextField))
    autoComplete="name"
    autoFocus={true}
    fullWidth={true}
    id="name"
    label="Name"
    name="name"
    onChange={[Function]}
    required={true}
    variant="outlined"
  />,
  <WithStyles(ForwardRef(TextField))
    autoComplete="email"
    fullWidth={true}
    id="email"
    label="Email Address"
    name="email"
    onChange={[Function]}
    required={true}
    variant="outlined"
  />,
  <WithStyles(ForwardRef(TextField))
    autoComplete="current-password"
    fullWidth={true}
    id="password"
    label="Password"
    name="password"
    onChange={[Function]}
    required={true}
    type="password"
    variant="outlined"
  />,
]
`;
